### DISCUSSION TITLE
How to set Sandboxed Browser as default Windows OS browser?

### DISCUSSION THREAD

**Discussion by ghost** [sandboxie-plus/Sandboxie#3675] [Q&A Feedback]
How to set Sandboxed Browser as default Windows OS browser? AFAIK windows just links default registered browser via HTTP/HTTPS default file type linking, but how do I make it launch default browser from within Sandboxie?

**Comment by isaak654** [comment]
> How to set Sandboxed Browser as default Windows OS browser? 

There is no such setting in Sandboxie, but someone provided a registry tweak for Chromium browser: https://github.com/sandboxie-plus/Sandboxie/issues/1745#issuecomment-1284413049

> how do I make it launch default browser from within Sandboxie?

Sandboxie Plus: right-click on your sandbox -> Run -> Standard Applications > Default Web Browser
Sandboxie Classic: right-click on your sandbox -> Run Sandboxed -> Run Web Browser

You can also create specific shortcuts, see [Sandboxie-Plus Migration Guide](https://sandboxie-plus.github.io/sandboxie-docs/Content/PlusMigrationGuide.html#create-sandbox-shortcut) and [Start Command Line](https://sandboxie-plus.github.io/sandboxie-docs/Content/StartCommandLine.html).

**Comment by ghost** [comment]
Thank you!

**Comment by bastik-1001** [comment]
You can also use the [force folder](https://sandboxie-plus.github.io/sandboxie-docs/Content/ForceFolder.html) or [force process](https://sandboxie-plus.github.io/sandboxie-docs/Content/ForceProcess.html) feature for browsers (or other software) that is installed on the host.

This would also start the browser in a sandbox if some other program launches it.

**Comment by ghost** [comment]
Thanks. What doesn't work is the other way around. I can't get browser to launch the needed app if either default browser or the app are sandboxed. Default browser is used to login to the app and that works, but when default browser tries to launch the app via handles, the app doesn't start.

**Reply by bastik-1001** [reply]
(Were you able to set the label "more info needed" yourself? I could not set a label for a discussion I started.)

When this procedure, you describe, involves a UWP based app, which is not supported by Sandboxie, yet, this explains why it would not work. Do you want to tell us the app that is supposed to be started?

**Comment by ghost** [comment]
Here's the normal process:

0. Default browser is non-UWP
1. Launch some (non-UWP) VPN app
2. Some VPN app requests login and launches non-UWP default browser (one registered OS as default browser)
3. Non-UWP default browser request login credentials
4. After login credentials are entered, non-UWP default browser asks "Would you like to open some VPN app?"
5. User confirms desire of opening some VPN app
6. Some VPN app launches and logs in user

If either some VPN app of non-UWP default browser or both are launched sandboxed, then the process breaks after step 5 and never makes it to step 6.  Some VPN app manages to launch sandboxed non-UWP default browser (step 2) and yet default browser isn't able to open some VPN app (step 6 never takes place).  It may be an impossible situation because some VPN app spawns tunnel driver with kernel access.

However, if UWP browser is installed and Sanboxie is not used for anything, then Step 6 occurs without issues. UWP in itself provides some sandboxing. In other words, UWP-based sandboxing for default browser allows for step 6 to happen, but Sandboxie-based sandboxing for default browser breaks step 6.

**Comment by ghost** [comment]
I am closed this one because it is too specific to the VPN app I use, which also launches a background service and kernel driver. Other sandboxed apps, such as uTorrent or Foxit PDF Reader, do not have a problem interacting with sandboxed default browser when correctly sandboxed.

**Comment by ghost** [comment]
Re-opening. Launching VPN app from sandboxed default web browser results in VPN app being migrated into default web browser sandbox! How do I prevent that from happening? I need to make sure VPN app is never migrated into any sandbox and launches from its installed location.

**Reply by bastik-1001** [reply]
Normally it is desired that everything is contained within a sandbox, but sometimes it is good when it can be opened in another box, which is why there are settings for it.

You can use [breakout process](https://sandboxie-plus.github.io/sandboxie-docs/Content/BreakoutProcess.html) or [breakout folder](https://sandboxie-plus.github.io/sandboxie-docs/Content/BreakoutFolder.html) per sandbox to allow something to be started outside the sandbox, either to be captured by another sandbox or to run un-sandboxed.

**Reply by ghost** [reply]
It works! It actually works! Sandboxed browser interacting with nonsandboxed VPN app is still better than both of them not being sandboxed. Thank you so much being so patient and helpful!
